### PR TITLE
Add dual list selector guidelines

### LINKED ISSUE #2197: Create dual list guidelines

### PR DESCRIPTION
**Pull request by mmenestr**
closes #2197

### CODE REVIEW

File: packages/v4/src/content/design-guidelines/components/dual-list-selector/dual-list-selector.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+3. **Chosen items list:** The chosen items list is a list of items that users have selected and moved from the available items list. 
+4. **Selected item:** A selected item refers to the  visual state of the item when it has been selected by a user.
+5. **Arrows:** Arrows are buttons that give users the ability to move items from one list to another. The arrows will enable and disable based on user selections. Single arrows move only selected items. Double arrows move everything from one list to the other (in the direction of the arrow), whether the list contains manually selected items or not.
+6. **Number of items:** The number of items is an indicator of how many items are selected among the total items available in the list.
+7. **Filter:** The filter is a search field that allows users to filter the list of items.
+8. **Sorting (optional):** The sorting ability enables users to sort the list of items presented for easier scanning.
+9. **More actions (optional):** The more actions (kebab) menu  stores added actions, such as exporting.
+
+### Expandable dual list
```
Comment (mcarrano) [inline packages/v4/src/content/design-guidelines/components/dual-list-selector/dual-list-selector.md:40]: Based on the earlier discussion, I thought you were only going to reference things here that are different from the basic list.

File: packages/v4/src/content/design-guidelines/components/dual-list-selector/dual-list-selector.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+2. **Available items list:** list of items that users can select/choose from. In this example, the list is multi-leveled, which means that items are grouped within folders.
+3. **Chosen items list:**  list of items that users have chosen from the available items list and moved into this list. In this case, when one item is moved to the chosen items, the folder that it was in also gets displayed. In this example, “Yellow” still displays inside a “Colors” folder after it has been moved to the chosen items.
+4. **Selected item:** visual state of item when it has been selected by a user. When a folder has a mix of selected and unselected items, the checkbox should have a mixed state. When all items in a folder are selected, the folder checkbox should have a selected state.
+5. **Number of items:**  indicator of how many items are selected, among the total items available in the list. In expandable lists, <u>only non-folder items are included in the item count.</u>
+6. **Badge (optional):** number of items inside the folder
+
```
Comment (mcarrano) [inline packages/v4/src/content/design-guidelines/components/dual-list-selector/dual-list-selector.md:51]: There are 10 items called out in the image but only 6 items here.

**Review by mcarrano — CHANGES_REQUESTED** [review]
See my comments on the Expandable list section.

File: packages/v4/src/content/design-guidelines/components/dual-list-selector/dual-list-selector.md (hunk 1 above)
Comment (mmenestr) [reply packages/v4/src/content/design-guidelines/components/dual-list-selector/dual-list-selector.md:40]: @mcarrano You're right, I forgot to delete some of them. I'm thinking maybe I should point out the following things instead:

1. Folder
2. Item
3. Selected item
4. Number of items (to specify what counts as a numbered item here, as opposed to a basic list)
5. Badge

What do you think?

**Comment by mmenestr** [comment]
@mcarrano updated! please take another look

**Review by mcarrano — APPROVED** [review]
Looks good @mmenestr . I will merge.

**Pull request merged**